lib/BloqadeExpr/src/types.jl

```diff
@@ -281,6 +281,51 @@ end
 
 SumOfZ(n::Int) = SumOfZ(n, 1)
 
+
+# this subtype is a bit fishy here
+# There are some places where this 
+# helps overload some functions but 
+# for clarity it might make sense to 
+# to make this a unique type.
+struct RydbergHamiltonian <: AbstractTerm
+    Terms::AbstractBlock
+    C::Real
+    function RydbergHamiltonian(atom_positions, C, Ω, ϕ, Δ)
```

Comment:
Usually it's preferred to keep the default constructor and define the constructor from outside unless there's a check must happen on all the objects. So ideally no matter how RydbergHamiltonian is defined because no checks are needed for its constructor unlike other terms. Its constructor should be defined outside.

Reply:
OK